components/elm/src/main/pftvarcon.F90

```diff
@@ -1103,12 +1223,209 @@ subroutine pftconrd
        nppercropmax         = nwillowirrig         ! last prognostic perennial crop in list
     end if
 
+   !-------------------------------------------------------------------------------------------
+    ! default: for tree and shrub always 1, now hard-coded as following
+    woody(ndllf_evr_tmp_tree:nbrdlf_dcd_brl_tree) = 1
+    woody(nbrdlf_evr_shrub:nbrdlf_dcd_brl_shrub)  = 2
+
+    ! the following is initialized as 0 above for all PFT.
+    ! here the hard-coded values (or flags) for default ELM PFT physiology will be working as original
+    ! when not using those indexing of PFT orders anymore in other codes than here.
+    needleleaf(noveg+1:ndllf_dcd_brl_tree) = 1
+    graminoid(nc3_arctic_grass:nc4_grass) = 1
+    iscft(npcropmin:max(npcropmax,nppercropmax)) = 1
+    nfixer(nsoybean)       = 1
+    nfixer(nsoybeanirrig)  = 1
+
+    climatezone(ndllf_evr_tmp_tree)  = 2
+    climatezone(ndllf_evr_brl_tree)  = 3
+    climatezone(ndllf_dcd_brl_tree)  = 3
+    climatezone(nbrdlf_evr_trp_tree) = 1
+    climatezone(nbrdlf_evr_tmp_tree) = 2
+    climatezone(nbrdlf_dcd_trp_tree) = 1
+    climatezone(nbrdlf_dcd_tmp_tree) = 2
+    climatezone(nbrdlf_dcd_brl_tree) = 3
+    climatezone(nbrdlf_dcd_tmp_shrub)= 2
+    climatezone(nbrdlf_dcd_brl_shrub)= 3
+    climatezone(nc3_arctic_grass)    = 4
+    !-------------------------------------------------------------------------------------------
+
+
+
+   ! NOT default PFT file
+   else
+
+       ! not vegetated checking
+       noveg = -1
+
+       ! when user-defined PFTs, if crop included, it must be the default way:
+       ! cropts must be in one block and after nat-pft if 'create_crop_landunit' or 'use_crop' is true
+       npcropmin            = -1                   ! first prognostic crop
+       npcropmax            = -1                   ! last prognostic crop in list
+       nppercropmin         = -1                   ! first prognostic perennial crop
+       nppercropmax         = -1                   ! last prognostic perennial crop in list
+
+       numcft = 0
+       ncft   = -1
+       ncft0  = -1
+       noncropmax = 0
+       do i = 0, npft-1
+          if (crop(i)>=1 .or. percrop(i)>=1 .or. iscft(i)>=1) then
+              numcft = numcft + 1  ! includes generic_crop, while cft_size NOT (???? todo checking)
+
+              if(use_crop) then
+
+                 ! the following assumes that all crop pfts are in a block
+
+                 ! if 'generic crop' (crop=1) specifically flagged by iscft=0
+                 ! 'crop' will not be counted into prognostic
+                 if (crop(i)>=1 .and. iscft(i)==1) then
+                    npcropmax = i
+                    if(npcropmin<=0) npcropmin = i
+                 end if
+
+                 ! NOTE: there is a misunderstanding in pft physiology parameter file: 'perennial crop' IS NOT 'crop',
+                 !       but still counted into 'numcft'
+                 if(percrop(i)==1) then
```

Comment:
Does this mean that percrop has an iscft ==0?

Reply:
No, percrop and iscft are independent.